sunpy/database/tests/test_tables.py

```diff
@@ -144,8 +144,9 @@ def test_entries_from_fido_search_result(fido_search_result):
         wavemin=0.1, wavemax=30.4)
     # 2 entries from goes
     assert entries[56] == DatabaseEntry(
-        source='nasa', provider='sdac', physobs='irradiance',
-        fileid='https://umbra.nascom.nasa.gov/goes/fits/2012/go1520120101.fits',
+        source='noaa', provider='sdac', physobs='irradiance',
```

Comment:
The SDAC is no longer the provider if we are now pulling from NOAA

Reply:
good point!

Reply:
actually following up on this - I guess how can this be handled in the case where data before GOES 13 is pulled from a NASA server, but data of > GOES 13 is now from the NOAA site?

Reply:
Oh right. Would it be possible to add the GOES < 13 (and the old calibration for 13-15) as a separate client? And then toggle between the old and new client based on the version number and satellite number attrs? I'm not sure if what I'm asking really makes any sense

Reply:
yeah that could make sense - however with this it would mean that the user would have to put in the satellite number for looking for the data which they may not know. At the moment if a user does 
` res = Fido.search(a.Time(tstart, tend), a.Instrument('XRS')`
and no satellite number is given it will find the available satellite that was taking data over that time and return that. 

Actually now that I think about it - how could you have two clients that both have a.Instrument('XRS') - would that work?

Reply:
oh actually I wonder if we could just have the [`_makeimap`](https://github.com/sunpy/sunpy/blob/master/sunpy/net/dataretriever/sources/goes.py#L158) call the [_get_url_for_timerange](https://github.com/sunpy/sunpy/blob/master/sunpy/net/dataretriever/sources/goes.py#L110) and check to see the url and then populate the server/source accordingly 

that will probably make more sense to do...